### DISCUSSION TITLE
Always dynamic pages in next build without i18n routing

### DISCUSSION THREAD

**Discussion by andrewLapshov** [amannn/next-intl#1542] [Q&A] [answered]
Hi! I recently started using next-intl and decided to try the without-routing implementation. However, when building the application, all routes are generated dynamically, even those that only contain static components. This can be reproduced with the example from the [documentation](https://github.com/amannn/next-intl/tree/main/examples/example-app-router-without-i18n-routing). Is it possible to get statically generated routes using the without-routing approach?

**Comment by amannn** [comment, marked as answer]
Maybe [`localePrefix: 'never'`](https://next-intl-docs.vercel.app/docs/routing#locale-prefix-never) could be useful for you? 

What is your use case btw.? Would be curious to learn a bit more why static rendering is relevant for you for a setup that doesn't require i18n routing. Is it a public-facing app or one behind a login or similar?

**Reply by andrewLapshov** [reply]
Thanks for the answer! Yes, this is a public-faced app, similar sites do not use router prefixes, so I also decided to do it without it. But your advice with localePrefix: 'never' seems to help me.
